src/power_grid_model_io/utils/auto_id.py

```diff
@@ -119,4 +119,4 @@ def __getitem__(self, idx: int) -> Any:
         Returns:
             The original item
         """
-        return self._items[idx]
+        return getattr(self._items[idx], None)
```

Comment:
What is the purpose of this change? pytest fails this one and passes when reverted to the original. (It should also be supplied with a `str` for the name field)

Reply:
i indeed think that this is not what you intended to do. as it is now, besides the missing `str` type, it will basically try to seek the attribute `None` in `self._items[idx]`, e.g. `self._items[idx].None`

My guess is that you want to make it such that you return `None` if `idx` is not in `self_items`? in that case, you will want to use

```suggestion
        return self._items.get(idx)
```

which is shorthand for 

```python
        return self._items.get(idx, None)
```

(the `None` is the default argument)